File: old/tags/2.2/java/edu/nps/moves/dis/IffFundamentalData.java
package edu.nps.moves.dis;

import java.util.*;
import java.io.*;
import edu.nps.moves.jaxb.dis.*;

/**
 * 5.2.42. Basic operational data ofr IFF ATC NAVAIDS
 *
 * Copyright (c) 2007, MOVES Institute, Naval Postgraduate School. All rights reserved.
 * This work is licensed under the BSD open source license, available at https://www.movesinstitute.org/licenses/bsd.html
 *
 * @author DMcG
 */
public class IffFundamentalData extends Object implements Serializable
{
   /** system status */
   protected short  systemStatus;

   /** Alternate parameter 4 */
   protected short  alternateParameter4;

   /** eight boolean fields */
   protected short  informationLayers;

   /** enumeration */
   protected short  modifier;

   /** parameter, enumeration */
   protected int  parameter1;

   /** parameter, enumeration */
   protected int  parameter2;

   /** parameter, enumeration */
   protected int  parameter3;

   /** parameter, enumeration */
   protected int  parameter4;

   /** parameter, enumeration */
   protected int  parameter5;

   /** parameter, enumeration */
   protected int  parameter6;


/** Constructor */
 public IffFundamentalData()
 {
 }

/** 
 * Constructor--takes a parallel jaxb object and returns an open-dis object 
 * 1.4_sed_bait_start */
 public IffFundamentalData(edu.nps.moves.jaxb.dis.IffFundamentalData x)
 {
     this.systemStatus = x.getSystemStatus();
     this.alternateParameter4 = x.getAlternateParameter4();
     this.informationLayers = x.getInformationLayers();
     this.modifier = x.getModifier();
     this.parameter1 = x.getParameter1();
     this.parameter2 = x.getParameter2();
     this.parameter3 = x.getParameter3();
     this.parameter4 = x.getParameter4();
     this.parameter5 = x.getParameter5();
     this.parameter6 = x.getParameter6();
 }
/* 1.4_sed_bait_end */


/**
 * returns a jaxb object intialized from this object, given an empty jaxb object
 * 1.4_sed_bait_start **/
 public edu.nps.moves.jaxb.dis.IffFundamentalData initializeJaxbObject(edu.nps.moves.jaxb.dis.IffFundamentalData x)
 {
     ObjectFactory factory = new ObjectFactory();

     x.setSystemStatus( this.getSystemStatus() );
     x.setAlternateParameter4( this.getAlternateParameter4() );
     x.setInformationLayers( this.getInformationLayers() );
     x.setModifier( this.getModifier() );
     x.setParameter1( this.getParameter1() );
     x.setParameter2( this.getParameter2() );
     x.setParameter3( this.getParameter3() );
     x.setParameter4( this.getParameter4() );
     x.setParameter5( this.getParameter5() );
     x.setParameter6( this.getParameter6() );
   return x;
 }
/* 1.4_sed_bait_end */


public int getMarshalledSize()
{
   int marshalSize = 0; 

   marshalSize = marshalSize + 1;  // systemStatus
   marshalSize = marshalSize + 1;  // alternateParameter4
   marshalSize = marshalSize + 1;  // informationLayers
   marshalSize = marshalSize + 1;  // modifier
   marshalSize = marshalSize + 2;  // parameter1
   marshalSize = marshalSize + 2;  // parameter2
   marshalSize = marshalSize + 2;  // parameter3
   marshalSize = marshalSize + 2;  // parameter4
   marshalSize = marshalSize + 2;  // parameter5
   marshalSize = marshalSize + 2;  // parameter6

   return marshalSize;
}


public void setSystemStatus(short pSystemStatus)
{ systemStatus = pSystemStatus;
}

public short getSystemStatus()
{ return systemStatus; 
}

public void setAlternateParameter4(short pAlternateParameter4)
{ alternateParameter4 = pAlternateParameter4;
}

public short getAlternateParameter4()
{ return alternateParameter4; 
}

public void setInformationLayers(short pInformationLayers)
{ informationLayers = pInformationLayers;
}

public short getInformationLayers()
{ return informationLayers; 
}

public void setModifier(short pModifier)
{ modifier = pModifier;
}

public short getModifier()
{ return modifier; 
}

public void setParameter1(int pParameter1)
{ parameter1 = pParameter1;
}

public int getParameter1()
{ return parameter1; 
}

public void setParameter2(int pParameter2)
{ parameter2 = pParameter2;
}

public int getParameter2()
{ return parameter2; 
}

public void setParameter3(int pParameter3)
{ parameter3 = pParameter3;
}

public int getParameter3()
{ return parameter3; 
}

public void setParameter4(int pParameter4)
{ parameter4 = pParameter4;
}

public int getParameter4()
{ return parameter4; 
}

public void setParameter5(int pParameter5)
{ parameter5 = pParameter5;
}

public int getParameter5()
{ return parameter5; 
}

public void setParameter6(int pParameter6)
{ parameter6 = pParameter6;
}

public int getParameter6()
{ return parameter6; 
}


public void marshal(DataOutputStream dos)
{
    try 
    {
       dos.writeByte( (byte)systemStatus);
       dos.writeByte( (byte)alternateParameter4);
       dos.writeByte( (byte)informationLayers);
       dos.writeByte( (byte)modifier);
       dos.writeShort( (short)parameter1);
       dos.writeShort( (short)parameter2);
       dos.writeShort( (short)parameter3);
       dos.writeShort( (short)parameter4);
       dos.writeShort( (short)parameter5);
       dos.writeShort( (short)parameter6);
    } // end try 
    catch(Exception e)
    { 
      System.out.println(e);}
    } // end of marshal method

public void unmarshal(DataInputStream dis)
{
    try 
    {
       systemStatus = dis.readByte();
       alternateParameter4 = dis.readByte();
       informationLayers = dis.readByte();
       modifier = dis.readByte();
       parameter1 = dis.readShort();
       parameter2 = dis.readShort();
       parameter3 = dis.readShort();
       parameter4 = dis.readShort();
       parameter5 = dis.readShort();
       parameter6 = dis.readShort();
    } // end try 
   catch(Exception e)
    { 
      System.out.println(e); 
    }
 } // end of unmarshal method 


 /**
  * The equals method doesn't always work--mostly on on classes that consist only of primitives. Be careful.
  */
 public boolean equals(IffFundamentalData rhs)
 {
     boolean ivarsEqual = true;

    if(rhs.getClass() != this.getClass())
        return false;

     if( ! (systemStatus == rhs.systemStatus)) ivarsEqual = false;
     if( ! (alternateParameter4 == rhs.alternateParameter4)) ivarsEqual = false;
     if( ! (informationLayers == rhs.informationLayers)) ivarsEqual = false;
     if( ! (modifier == rhs.modifier)) ivarsEqual = false;
     if( ! (parameter1 == rhs.parameter1)) ivarsEqual = false;
     if( ! (parameter2 == rhs.parameter2)) ivarsEqual = false;
     if( ! (parameter3 == rhs.parameter3)) ivarsEqual = false;
     if( ! (parameter4 == rhs.parameter4)) ivarsEqual = false;
     if( ! (parameter5 == rhs.parameter5)) ivarsEqual = false;
     if( ! (parameter6 == rhs.parameter6)) ivarsEqual = false;

    return ivarsEqual;
 }
} // end of class

File: old/tags/2.2/cpp/DIS/IffFundamentalData.cpp
#include <DIS/IffFundamentalData.h> 

using namespace DIS;


IffFundamentalData::IffFundamentalData():
   _systemStatus(0), 
   _alternateParameter4(0), 
   _informationLayers(0), 
   _modifier(0), 
   _parameter1(0), 
   _parameter2(0), 
   _parameter3(0), 
   _parameter4(0), 
   _parameter5(0), 
   _parameter6(0)
{
}

IffFundamentalData::~IffFundamentalData()
{
}

unsigned char IffFundamentalData::getSystemStatus() const
{
    return _systemStatus;
}

void IffFundamentalData::setSystemStatus(unsigned char pX)
{
    _systemStatus = pX;
}

unsigned char IffFundamentalData::getAlternateParameter4() const
{
    return _alternateParameter4;
}

void IffFundamentalData::setAlternateParameter4(unsigned char pX)
{
    _alternateParameter4 = pX;
}

unsigned char IffFundamentalData::getInformationLayers() const
{
    return _informationLayers;
}

void IffFundamentalData::setInformationLayers(unsigned char pX)
{
    _informationLayers = pX;
}

unsigned char IffFundamentalData::getModifier() const
{
    return _modifier;
}

void IffFundamentalData::setModifier(unsigned char pX)
{
    _modifier = pX;
}

unsigned short IffFundamentalData::getParameter1() const
{
    return _parameter1;
}

void IffFundamentalData::setParameter1(unsigned short pX)
{
    _parameter1 = pX;
}

unsigned short IffFundamentalData::getParameter2() const
{
    return _parameter2;
}

void IffFundamentalData::setParameter2(unsigned short pX)
{
    _parameter2 = pX;
}

unsigned short IffFundamentalData::getParameter3() const
{
    return _parameter3;
}

void IffFundamentalData::setParameter3(unsigned short pX)
{
    _parameter3 = pX;
}

unsigned short IffFundamentalData::getParameter4() const
{
    return _parameter4;
}

void IffFundamentalData::setParameter4(unsigned short pX)
{
    _parameter4 = pX;
}

unsigned short IffFundamentalData::getParameter5() const
{
    return _parameter5;
}

void IffFundamentalData::setParameter5(unsigned short pX)
{
    _parameter5 = pX;
}

unsigned short IffFundamentalData::getParameter6() const
{
    return _parameter6;
}

void IffFundamentalData::setParameter6(unsigned short pX)
{
    _parameter6 = pX;
}

void IffFundamentalData::marshal(DataStream& dataStream) const
{
    dataStream << _systemStatus;
    dataStream << _alternateParameter4;
    dataStream << _informationLayers;
    dataStream << _modifier;
    dataStream << _parameter1;
    dataStream << _parameter2;
    dataStream << _parameter3;
    dataStream << _parameter4;
    dataStream << _parameter5;
    dataStream << _parameter6;
}

void IffFundamentalData::unmarshal(DataStream& dataStream)
{
    dataStream >> _systemStatus;
    dataStream >> _alternateParameter4;
    dataStream >> _informationLayers;
    dataStream >> _modifier;
    dataStream >> _parameter1;
    dataStream >> _parameter2;
    dataStream >> _parameter3;
    dataStream >> _parameter4;
    dataStream >> _parameter5;
    dataStream >> _parameter6;
}


bool IffFundamentalData::operator ==(const IffFundamentalData& rhs) const
 {
     bool ivarsEqual = true;

     if( ! (_systemStatus == rhs._systemStatus) ) ivarsEqual = false;
     if( ! (_alternateParameter4 == rhs._alternateParameter4) ) ivarsEqual = false;
     if( ! (_informationLayers == rhs._informationLayers) ) ivarsEqual = false;
     if( ! (_modifier == rhs._modifier) ) ivarsEqual = false;
     if( ! (_parameter1 == rhs._parameter1) ) ivarsEqual = false;
     if( ! (_parameter2 == rhs._parameter2) ) ivarsEqual = false;
     if( ! (_parameter3 == rhs._parameter3) ) ivarsEqual = false;
     if( ! (_parameter4 == rhs._parameter4) ) ivarsEqual = false;
     if( ! (_parameter5 == rhs._parameter5) ) ivarsEqual = false;
     if( ! (_parameter6 == rhs._parameter6) ) ivarsEqual = false;

    return ivarsEqual;
 }

int IffFundamentalData::getMarshalledSize() const
{
   int marshalSize = 0;

   marshalSize = marshalSize + 1;  // _systemStatus
   marshalSize = marshalSize + 1;  // _alternateParameter4
   marshalSize = marshalSize + 1;  // _informationLayers
   marshalSize = marshalSize + 1;  // _modifier
   marshalSize = marshalSize + 2;  // _parameter1
   marshalSize = marshalSize + 2;  // _parameter2
   marshalSize = marshalSize + 2;  // _parameter3
   marshalSize = marshalSize + 2;  // _parameter4
   marshalSize = marshalSize + 2;  // _parameter5
   marshalSize = marshalSize + 2;  // _parameter6
    return marshalSize;
}

